components/core/src/clp_s/CMakeLists.txt

```diff
@@ -0,0 +1,141 @@
+        FileReader.hpp
+        FileWriter.cpp
+        FileWriter.hpp
+        JsonConstructor.cpp
+        JsonConstructor.hpp
+        JsonFileIterator.cpp
+        JsonFileIterator.hpp
+        JsonParser.cpp
+        JsonParser.hpp
+        JsonSerializer.cpp
+        JsonSerializer.hpp
+        ParsedMessage.cpp
+        ParsedMessage.hpp
+        ReaderUtils.cpp
+        ReaderUtils.hpp
+        SchemaMap.cpp
+        SchemaMap.hpp
+        SchemaTree.cpp
+        SchemaTree.hpp
+        SchemaReader.cpp
+        SchemaReader.hpp
+        SchemaWriter.cpp
+        SchemaWriter.hpp
+        TimestampDictionaryReader.cpp
+        TimestampDictionaryReader.hpp
+        TimestampDictionaryWriter.cpp
+        TimestampDictionaryWriter.hpp
+        TimestampEntry.cpp
+        TimestampEntry.hpp
+        TimestampPattern.cpp
+        TimestampPattern.hpp
+        TraceableException.cpp
+        TraceableException.hpp
+        Utils.cpp
+        Utils.hpp
+        VariableDecoder.cpp
+        VariableDecoder.hpp
+        VariableEncoder.cpp
+        VariableEncoder.hpp
+        ZstdCompressor.cpp
+        ZstdCompressor.hpp
+        ZstdDecompressor.cpp
+        ZstdDecompressor.hpp
+        "${PROJECT_SOURCE_DIR}/submodules/date/include/date/date.h"
+)
+
+set(CLP_S_SEARCH_SOURCES
+        search/AndExpr.cpp
+        search/AndExpr.hpp
+        search/BooleanLiteral.cpp
+        search/BooleanLiteral.hpp
+        search/ColumnDescriptor.cpp
+        search/ColumnDescriptor.hpp
+        search/ConstantProp.cpp
+        search/ConstantProp.hpp
+        search/ConvertToExists.cpp
+        search/ConvertToExists.hpp
+        search/DateLiteral.hpp
+        search/DateLiteral.cpp
+        search/EmptyExpr.cpp
+        search/EmptyExpr.hpp
+        search/EvaluateTimestampIndex.cpp
+        search/EvaluateTimestampIndex.hpp
+        search/Expression.cpp
+        search/Expression.hpp
+        search/FilterExpr.cpp
+        search/FilterExpr.hpp
+        search/FilterOperation.hpp
+        search/Integral.cpp
+        search/Integral.hpp
+        search/Literal.hpp
+        search/NarrowTypes.cpp
+        search/NarrowTypes.hpp
+        search/NullLiteral.cpp
+        search/NullLiteral.hpp
+        search/OrExpr.cpp
+        search/OrExpr.hpp
+        search/OrOfAndForm.cpp
+        search/OrOfAndForm.hpp
+        search/Output.cpp
+        search/Output.hpp
+        search/SchemaMatch.cpp
+        search/SchemaMatch.hpp
+        search/SearchUtils.cpp
+        search/SearchUtils.hpp
+        search/StringLiteral.cpp
+        search/StringLiteral.hpp
+        search/Transformation.hpp
+        search/Value.hpp
+        search/clp_search/EncodedVariableInterpreter.cpp
+        search/clp_search/EncodedVariableInterpreter.hpp
+        search/clp_search/Grep.cpp
+        search/clp_search/Grep.hpp
+        search/clp_search/Query.cpp
+        search/clp_search/Query.hpp
+)
+
+add_executable(clp_s clp_s.cpp ${CLP_S_SOURCES} ${CLP_S_SEARCH_SOURCES})
```

Comment:
Let's change the executable name to `clp-s` so it's easier to type.

Reply:
Should we also change the directory and namespace name?

Reply:
Nah, namespace name needs to use underscores (don't think C++ supports hyphens in the namespace name). And since the directory name needs to match the namespace name, it should also use underscores.